src/lib/rnp.cpp

```diff
@@ -2665,6 +2665,23 @@ rnp_verify_dest_provider(pgp_parse_handler_t *handler,
     return true;
 }
 
+static void
+rnp_verify_on_decryption_info(bool mdc, pgp_aead_alg_t aead, pgp_symm_alg_t salg, void *param)
+{
+    rnp_op_verify_t op = (rnp_op_verify_t) param;
```

Comment:
@rrrooommmaaa This is a common practice to add some (void *) param to context information, so caller, who uses this context, may add his own information (like rnp_op_verify_t structure). I.e. lower-layer code (like `init_encrypted_src()` and others) knows nothing about the type, but caller knows the param type and may cast it in callback to correctly process received data.

Reply:
I think this is a pretty standard `void *userdata` pattern for callbacks. Although we could do fancier C++ things internally, ultimately we'll end up using `void*` for the FFI. We could probably do something with RTTI or some magic marker, but whether it would be worth it is another question.